docs/devcontainerjson-reference.md

```diff
@@ -0,0 +1,172 @@
+# devcontainer.json reference
+
+A `devcontainer.json` file in your project tells `devcontainer.json` supporting services / tools how to access (or create) a **development container** with a well-defined tool and runtime stack. It's currently supported by the [Remote - Containers](https://aka.ms/vscode-remote/download/containers) extension and [GitHub Codespaces](https://github.com/features/codespaces).
```

Comment:
Maybe "tools / services that support the dev container spec" rather than "`devcontainer.json` supporting services / tools"?

Reply:
Also, should we list the tools / services that support the spec here, or move those to somewhere else like a welcome page?

Reply:
Linking to another page upon first mention of `tools / services that support the dev container spec` seems useful (as this page is already quite long).

Do you think we have sufficient content to make that page now?

Reply:
I think so. We can likely document the open source reference implementation, the @vscode/dev-container-cli, Remote - Containers, and GitHub Codespaces.

In fact, we could move tool/service specific properties and namespaces in this same location (e.g. the two VS Code properties). The LSP site mentions that people can PR their content - it seems reasonable that we'd allow others to do the same or just link to their own reference documentation if they prefer.

Reply:
Thanks for the insight. I added a `supporting-tools.md` file and updated the reference doc's intro. Let me know what you think.